hostap: Security improvements (#74847)
zephyrproject-rtos/zephyr: doc/connectivity/networking/api/wifi.rst

* Networking or socket offloaded drivers
* Native L2 Ethernet drivers

Wi-Fi Enterprise test: X.509 Certificate header generation
##########################################################

Wi-Fi enterprise security requires use of X.509 certificates, test certificates
in PEM format are committed to the repo and the below steps to be followed to convert
them to C header file that can be included in Wi-Fi shell module.

Convert CA format from PEM to DER
=======================================
To convert the CA format from PEM to DER, use the following command:

> .. code-block:: none

Reviewer: This isn't being rendered properly, needs a newline before as well.

Author: Hi @krish2718 thanks, please review again

Reviewer: FYI, when you do a force push after resolving comments, reviewers get an email, and you can still trigger a specific email for a reviewer by clicking on `refresh` button as below.

![image](https://github.com/user-attachments/assets/425bdc8e-0ec6-4796-9d31-dde95ec19508)

Author: thanks for your info

* Networking or socket offloaded drivers
* Native L2 Ethernet drivers

Wi-Fi Enterprise test: X.509 Certificate header generation

> ##########################################################

Reviewer: Sorry, I gave you wrong heading format. This should be as per https://docs.zephyrproject.org/latest/contribute/documentation/guidelines.html#headings 

```suggestion
**************************************************
```

Wi-Fi enterprise security requires use of X.509 certificates, test certificates
in PEM format are committed to the repo and the below steps to be followed to convert
them to C header file that can be included in Wi-Fi shell module.

Convert CA format from PEM to DER

> =======================================

Reviewer: ```suggestion
==============================
```

Heading format should align till end of the text

*************

.. doxygengroup:: wifi_mgmt

> .. _wifi_shell_test_certs:

Author: Hi @krish2718 how to change this?
.. _wifi_shell_test_certs:

> .. _wifi_shell_test_certs:

Author: Hi @krish2718 
please review again

X.509 Certificate header generation
###################################

Wi-Fi enterprise security requires use of X.509 certificates, test certificates
in PEM format are committed to the repo and the below steps to be followed to convert
them to C header file that can be included in Wi-Fi shell module.

Convert CA format from PEM to DER
=================================
To convert the CA format from PEM to DER, use the following command:

> .. code-block:: none
> openssl x509 -outform der -in cas.pem -out ca.der

Reviewer: This needs a specific format:
```suggestion
.. code-block:: none

     openssl x509 -outform der -in cas.pem -out ca.der
```
See for an example https://raw.githubusercontent.com/zephyrproject-rtos/zephyr/main/samples/net/wifi/README.rst

Convert RSA Key format to DER
=============================
To convert the RSA Key format to DER, use the following command:
.. code-block:: none
openssl rsa -in wifiuser.key -outform DER -out client_key.der

Convert .der to header file
===========================
To convert the .der file to a header file, use the following command:
.. code-block:: none
xxd -i ca.der > ca.h

Replace corresponding cert header files
=======================================
Replace the corresponding cert header files at the following path:
subsys/net/l2/wifi/test_certs/

> .. note:: Please ensure that the parameter names are the same.

Reviewer: Format to be fixed.

```suggestion
.. note:: 

   Please ensure that the parameter names in the header are unchanged, only the payload is updated.
```

*************

.. doxygengroup:: wifi_mgmt

> X.509 Certificate header generation

Reviewer: Looking at https://builds.zephyrproject.io/zephyr/pr/74847/docs/connectivity/networking/api/wifi.html

this should be before the doxygen docs i.e., before `API reference`

> X.509 Certificate header generation

Reviewer: ```suggestion
Wi-Fi Enterprise test: X.509 Certificate header generation
```

Author: Hi @krish2718 
please review again

Author: Hi @krish2718 @jukkar  please help review again
